witchcraft-server/src/body.rs

```diff
@@ -44,47 +45,56 @@ impl RequestBody {
         RequestBody {
             inner,
             cur: Bytes::new(),
+            trailers: None,
             _p: PhantomPinned,
         }
     }
-
     /// Returns the request's trailers, if any are present.
     ///
     /// The body must have been completely read before this is called.
-    pub fn poll_trailers(
+    pub fn trailers(self: Pin<&mut Self>) -> Option<HeaderMap> {
+        self.project().trailers.take()
+    }
+
+    fn poll_next_raw(
         self: Pin<&mut Self>,
         cx: &mut Context<'_>,
-    ) -> Poll<Result<Option<HeaderMap>, Error>> {
-        self.project()
-            .inner
-            .poll_trailers(cx)
-            .map_err(|e| Error::service_safe(e, ClientIo))
-    }
+    ) -> Poll<Option<Result<Bytes, hyper::Error>>> {
+        let mut this = self.project();
```

Comment:
What's the difference between this and the `self.as_mut().project()` below https://github.com/palantir/witchcraft-rust-server/pull/158/files#diff-91aac5a936882d40f7bbd7ae1f10863422f0398383c828a40e5265922f6c08ffR91?

Reply:
The as_mut allows the `Pin<&mut Self>` to be accessed multiple times. `self.project()` will move out of `self`, while `self.as_mut().project()` won't: https://doc.rust-lang.org/stable/std/pin/struct.Pin.html#method.as_mut